### PR TITLE
Moving Clipboard related tests to a new class and combine the ClipboardTests class

### PR DESCRIPTION
**Pull request by LeafShi1**
Related #11948

## Proposed changes

- Moving the tests that using Clipboard under class `RichTextBoxTests` to a new class `RichTextBoxTests.ClipboardTests.cs`
- Combine the class `ClipboardTests.cs`

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11936)

### CODE REVIEW

**Review by ricardobossan — COMMENTED** [review]
LGTM

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good to me but I'd like @lonitra to review it. I'm not sure if ClipboardComTests are still relevant.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ClipboardComTests.cs [hunk 1]
```diff
@@ -1,15 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
-namespace System.Windows.Forms.Tests;
-
-public partial class ClipboardTests
-{
-    [WinFormsFact]
-    public void Clipboard_SetText_InvokeString_GetReturnsExpected()
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ClipboardComTests.cs:9]: This needs to be kept here. This is testing that this behavior works as expected with BuiltInComSupport is turned off, which is the case for trimming scenarios.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ClipboardTests.cs [hunk 2]
```diff
@@ -16,8 +16,16 @@ namespace System.Windows.Forms.Tests;
...
 [Collection("Sequential")]
 [CollectionDefinition("Sequential", DisableParallelization = true)]
-public partial class ClipboardTests
+public class ClipboardTests
 {
+    [WinFormsFact]
+    public void Clipboard_SetText_InvokeString_GetReturnsExpected()
+    {
+        Clipboard.SetText("text");
+        Assert.Equal("text", Clipboard.GetText());
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ClipboardTests.cs:25]: Can we update this to use fluent assertions

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RichTextBoxTests.ClipboardTests.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+public partial class RichTextBoxTests
+{
+    [Collection("Sequential")]
+    public class ClipboardTests
+    {
+        [WinFormsFact]
+        public void RichTextBox_OleObject_IncompleteOleObject_DoNothing()
+        {
+            using RichTextBox control = new();
+            Assert.NotEqual(IntPtr.Zero, control.Handle);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RichTextBoxTests.ClipboardTests.cs:17]: Same here with fluent assertions

**Review by lonitra — APPROVED** [review]
Thank you

**Pull request merged**